routes/staff.js

```diff
@@ -90,4 +90,57 @@ router.get("/order-summary/:rid",
         return res.send(orderList);
     });
 
+router.get("/promos/:rid",
+    [
+        check('rid').isInt()
+    ],
+    validate
+    , async (req, res, next) => {
+        let promoList = [];
+        try {
+            promoList = await getRestaurantPromos(req.params.rid);
+        } catch (err) {
+            return next(err)
+        }
+        return res.send(promoList);
+    });
+
+router.post("/add-promos/",
+    [
+        check('rid').isInt()
```

Comment:
This allows staff to add to any restaurant as long as he knows their rid. Maybe don't give it as param but rather retrieve it in the backend instead.